src/components/posts/PostForm/PostForm.tsx

```diff
@@ -0,0 +1,74 @@
+import { Grid } from '@material-ui/core';
+import React from 'react';
+import {
+  displayPostShareability,
+  PostShareabilityList,
+  PostTypeList,
+} from '../../../api';
+import {
+  DialogForm,
+  DialogFormProps,
+} from '../../../components/Dialog/DialogForm';
+import {
+  FieldGroup,
+  SecuredField,
+  SelectField,
+  SubmitError,
+  TextField,
+} from '../../../components/form';
+import { minLength, required } from '../../form/validators';
+import { PostFormFragment } from './PostForm.generated';
+
+export type PostFormProps<T, R = void> = DialogFormProps<T, R> & {
+  /** The pre-existing user to edit */
+  post?: PostFormFragment;
+  prefix: string;
```

Comment:
Shouldn't prefix be static to the specific form?

Also DialogForm has a fieldsPrefix prop that could be used instead of the <FieldGroup> component.